fog/ledger/server/tests/router_connection.rs

```diff
@@ -863,6 +863,281 @@ fn fog_ledger_untrusted_tx_out_api_test(logger: Logger) {
     sleep(Duration::from_millis(1000));
 }
 
+// Test that a fog ledger connection is able to check key images by hitting
+// a fog ledger router using the unary API
+#[test_with_logger]
+fn fog_router_unary_key_image_test(logger: Logger) {
+    let mut rng = RngType::from_seed([0u8; 32]);
+
+    for block_version in BlockVersion::iterator() {
```

Comment:
Can we use the `yare` crate to parametrize this test over `BlockVersion` rather than do a loop? That way, it's more easily configurable and we when one of the tests fails, there will be a more descriptive test name that incorporates the block version. 

See https://github.com/mobilecoinfoundation/mobilecoin/blob/feature/fog-view-router/fog/view/server/tests/streaming_smoke_tests.rs#L14

Reply:
I think this falls under #3168